### PR TITLE
Implement linear solver for least squares

### PR DESCRIPTION
**Pull request by ddement**
This changes our TPL requirements.  We now require our Kokkos-Kernels to be configured to use Lapack, which gives us access to Lapack's gesv function under the hood.

### CODE REVIEW

File: .github/workflows/correctness-linux.yaml [hunk 1]
```diff
@@ -31,7 +31,7 @@ jobs:
         spack install googletest
         spack install cppcheck
         spack install yaml-cpp
-        spack install trilinos@master~epetra
+        spack install trilinos@master~epetra ^kokkos-kernels+blas+lapack
```
Comment (faisal-bhuiyan) [inline .github/workflows/correctness-linux.yaml:34]: Could you add this to the build recipes?

File: src/beams/interpolation.hpp [hunk 2]
```diff
@@ -166,7 +166,8 @@ inline std::vector<double> GenerateGLLPoints(const size_t order) {
     }
 
     const size_t n_nodes = order + 1;
-    std::vector<double> gll_points(n_nodes, 0.);
+    auto gll_points = std::vector<double>(n_nodes, 0.);
+    gll_points.resize(n_nodes);
```
Comment (faisal-bhuiyan) [inline src/beams/interpolation.hpp:170]: Is this resizing necessary here, since we already initialize the `std::vector`?

**Review by faisal-bhuiyan — APPROVED** [review]
Looks good

**Review by deslaughter — APPROVED** [review]

File: .github/workflows/correctness-linux.yaml (hunk 1 above)
Comment (ddement) [reply .github/workflows/correctness-linux.yaml:34]: Will do

File: src/beams/interpolation.hpp (hunk 2 above)
Comment (ddement) [reply src/beams/interpolation.hpp:170]: It shouldn't be.  In fact, it's a no-op.  I was getting compiler warnings locally (gcc 12.3) from the `.front()` and `.back()` calls saying that it was dereferencing a null pointer.  Obviously, the warnings were incorrect, but calling resize here silences them.  I wish I knew a better way.

**Pull request merged**